numpy/core/src/multiarray/mapping.c

```diff
@@ -498,7 +498,10 @@ prepare_index(PyArrayObject *self, PyObject *index,
             PyArrayObject *tmp_arr;
             tmp_arr = (PyArrayObject *)PyArray_FROM_O(obj);
             if (tmp_arr == NULL) {
-                /* TODO: Should maybe replace the error here? */
+                PyErr_SetString(PyExc_IndexError,
+                        "only integers, slices (`:`), ellipsis (`...`), "
+                        "numpy.newaxis (`None`) and integer or boolean "
+                        "arrays are valid indices");
```

Comment:
This was a bugfix for the case `arr = np.arange(10);  arr[((Ellipsis,)*2,)]`. Numpy tries to convert that index to an array in line 499, which fails with a `ValueError` now (see above), but we should actually make it an `IndexError`.

Reply:
Is this fix independent of the rest of this patch?

Reply:
Yeah I think so. Although in numpy master this branch almost never gets hit, since the only way to hit it is with an object that can't be converted to an object array, which is rare.